### PR TITLE
Remove forkchoice call from notify new payload

### PR DESCRIPTION
**Pull request by potuz**
This PR although simple does a few things
- It removes the handing of invalid blockhash error, cf https://github.com/ethereum/execution-apis/issues/270
- It removes the handling of invalid payloads from the call to newPayload this avoids a write lock on forkchoice on this function which is necessary to parallelize execution and consenus validation. 

It adds the lvh member to the invalid block structure.

### CODE REVIEW

File: beacon-chain/blockchain/process_block.go [hunk 1]
```diff
@@ -134,6 +135,9 @@ func (s *Service) onBlock(ctx context.Context, signed interfaces.ReadOnlySignedB
 	}
 	isValidPayload, err := s.notifyNewPayload(ctx, postStateVersion, postStateHeader, signed)
 	if err != nil {
+		if IsInvalidBlock(err) && InvalidBlockLVH(err) != [32]byte{} {
+			return s.reportInvalidBlock(ctx, blockRoot, parentRoot, InvalidBlockLVH(err))
```
Comment (terencechain) [inline beacon-chain/blockchain/process_block.go:139]: I think you are losing the error message here

File: beacon-chain/blockchain/execution_engine.go [hunk 2]
```diff
@@ -220,35 +220,36 @@ func (s *Service) notifyNewPayload(ctx context.Context, postStateVersion int,
...
-	case execution.ErrInvalidBlockHashPayloadStatus:
-		newPayloadInvalidNodeCount.Inc()
-		return false, ErrInvalidBlockHashPayloadStatus
 	default:
 		return false, errors.WithMessage(ErrUndefinedExecutionEngineError, err.Error())
 	}
 }
 
+// reportInvalidBlock deals with the event that an invalid block was detected by the execution layer
+func (s *Service) reportInvalidBlock(ctx context.Context, root, parentRoot, lvh [32]byte) error {
```
Comment (terencechain) [inline beacon-chain/blockchain/execution_engine.go:233]: We probably want a unit test since this is likely uncovered.

**Review by terencechain — CHANGES_REQUESTED** [review]

File: beacon-chain/blockchain/process_block.go (hunk 1 above)
Comment (potuz) [reply beacon-chain/blockchain/process_block.go:139]: It will return the error `ErrInvalidPayload` which contains the relevant message

Comment (terencechain) [reply beacon-chain/blockchain/process_block.go:139]: You are right. I missed that field

**Review by terencechain — APPROVED** [review]

**Pull request merged**